### ISSUE TITLE
Using path aliases in cra

### ISSUE THREAD

**Issue by ramandhingra** [NiGhTTraX/ts-monorepo#87]
Hi @NiGhTTraX 

In the cra app, I want to use the path aliases which, if I am right, could be specified in the cra's tsconfig.json as follows:

```
  "extends": "../../tsconfig.json", 
    ...
    baseUrl: "src",
    "paths": {
      "@contexts/*": ["contexts/*"],
      "@hooks/*": ["hooks/*"],
      "@shared/*": ["shared/*"],
    },

```

But this is not working. What do I need to do to make it work, any hints.

Kindly note that the ts-monorepo has its own ts-config file where the "paths" property is already there.  

Thank you.

**Comment by NiGhTTraX** [comment]
Hi @ramandhingra, did you check the [cra example](https://github.com/NiGhTTraX/ts-monorepo/tree/2ac3c239d655fbcd48a1ebc88f97f2fe54061929#create-react-app)?

**Comment by ramandhingra** [comment]
Hello @NiGhTTraX 

My bad.. the cra server was not running. I just added the baseUrl property in the cra's tsconfig.json and it is working as desired.

Thank you.

**Comment by ramandhingra** [comment]
Hello @NiGhTTraX 

I tried to resolve this using several ways including trying to configure using webpack-plugin but no answer. If I try to override the baseUrl and/or paths in the cra's tsconfig, then the paths (specified in the root tsconfig) to the packages/ directory get lost. 

So I want to access both of the following at the same time in the examples/cra web app:
a. packages/* like foo, bar, components
b. Make path aliases to the following directories: hooks, shared, contexts, utils lying under the examples/cra/src The need for this is to avoid using the relative paths like import dateUtils from ../../utils. Instead use the import .. from "utils/"

Your solution is already offering the (a) functionality. But (b) is not there.

Could you pl. help answer it.

**Comment by NiGhTTraX** [comment]
CRA doesn't officially support path aliases and will try to remove them from the `tsconfig.json` inside the project, but it won't remove them if they come from an extended config - that's why the aliases defined in the root of this monorepo project work.

To add aliases that are specific to your CRA app, you can duplicate the ones in the root, add your own and then set a common `baseUrl`, or, if you're running a recent version of TypeScript, remove the `baseUrl` from the root and set all aliases to be relative to the config they're defined in.

To summarize:
- have your CRA's `tsconfig.json` use `{ extends: "some-other-config.json" }`
- either set all the aliases in that config with one `baseUrl`, or remove the one in the root `tsconfig.json` and change all aliases to use `./`.

**Comment by ramandhingra** [comment]
Thanks for your tips. This is what I have done:

Created the tsconfig.paths.json in the CRA directory with the following contents. It has all the aliases a. packages/* (your code) b. CRA-specific aliases.

```
{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@my-project/components/*": ["../../packages/components/src/*"],
      "@my-project/*": ["../../packages/*/src"],
      "@constants/*": ["./src/constants/*"],
      "@hooks/*": ["./src/hooks/*"],
    },
  }
}


```

Then I extended the above file in the CRA's tsconfig.json meaning that the root tsconfig.json is bypassed in the CRA. When I run the CRA, it is able to recognise these aliases.

But the Intellisense is not working. It is unable to recognise the aliases and reports the errors, for example, Cannot find module "constants" or its corresponding type declarations typescript.

**Comment by NiGhTTraX** [comment]
@ramandhingra that looks correct. For comparison, I've updated the CRA package with a similar example: 985e2ff8912764f3be7cb7a0f17ff26a3161c27f.

> But the Intellisense is not working

You mean in your IDE? If CRA builds correctly, maybe the IDE hasn't picked up the updated configs and just needs a restart? The example that I've linked above works without any problems in Webstorm.

**Comment by ramandhingra** [comment]
Yes, the VS Code IDE. Restarting it made Intellisense work. Thank you.
